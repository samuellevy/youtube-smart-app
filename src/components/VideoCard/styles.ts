import styled from 'styled-components';
import { colors } from '../../global';

export const Container = styled.div``;
export const ThumbBox = styled.div`
  /* border-radius: 8px;
  border: 5px #fff solid; */
  padding: 7px;
`;
export const ThumbImage = styled.img`
  width: 28vw;
`;
export const Title = styled.h2``;
export const Author = styled.div`
  color: ${colors.gray};
  font-size: 14px;
`;
export const Footer = styled.div`
  color: ${colors.gray};
  font-size: 14px;
`;
export const Info = styled.div``;
