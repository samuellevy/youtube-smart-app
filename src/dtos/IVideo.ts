export interface IVideo {
  id: string;
  publishedAt: string;
  channelId: string;
  title: string;
  description: string;
  thumbnail: {
    default: {url:string},
    medium: {url: string;}
    high: {url:string}
  };
  channelTitle: string;
  viewCount: number;
}
