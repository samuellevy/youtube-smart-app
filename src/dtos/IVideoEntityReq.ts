export interface IVideoEntityReq {
  id: string,
  snippet: IVideoSnippet,
  statistics: {
    viewCount: number;
  }
}

interface IVideoSnippet {
  publishedAt: string;
  channelId: string;
  channelTitle: string;
  title: string;
  description: string;
  thumbnails: IVideoThumbnail;
}

interface IVideoThumbnail {
  high: {
    url: string;
  }
}
