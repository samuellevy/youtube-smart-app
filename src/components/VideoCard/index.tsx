import React from 'react';

import * as S from './styles';

interface IVideoCardComponent {
  data: IVideoCardItem;
  active?: boolean;
}

interface IVideoCardItem {
  title: string;
  author: string;
  views: string;
  publishDate: string;
  active?: boolean;
}

const VideoCard: React.FC<IVideoCardComponent> = ({ data, active }: IVideoCardComponent) => (
  <S.Container>
    <S.ThumbBox active={active}>
      <S.ThumbImage src="https://i.ytimg.com/vi/vJJX5gmP7oA/mqdefault.jpg" alt="" />
    </S.ThumbBox>
    <S.Title>{data.title}</S.Title>
    <S.Author>{data.author}</S.Author>
    <S.Footer>
      <S.Info>
        {`${data.views} views â€¢ ${data.publishDate}`}
      </S.Info>
    </S.Footer>
  </S.Container>
);

export default VideoCard;
