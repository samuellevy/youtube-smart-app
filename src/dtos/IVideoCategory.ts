import { IVideo } from './IVideo';

export interface IVideoCategory {
  id: string;
  title: string;
  videos: IVideo[];
}
