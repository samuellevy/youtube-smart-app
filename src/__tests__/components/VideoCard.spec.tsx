import React from 'react';
import Enzyme, { shallow, mount, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer, { ReactTestRendererJSON } from 'react-test-renderer';

import { fireEvent, waitFor } from '@testing-library/react';
import { title } from 'process';
import VideoCard from '../../components/VideoCard';
import { IVideo } from '../../dtos/IVideo';

Enzyme.configure({ adapter: new Adapter() });

const mockedHandleOut = jest.fn();
const mockedHandlePlay = jest.fn();
const mockedKeyboard = jest.fn();
const mockedDispatch = jest.fn();

const mockVideoItem: IVideo = {
  channelId: 'a', channelTitle: 'teste', description: 'description', id: 'id', publishedAt: 'published', thumbnail: { default: { url: '' }, high: { url: '' }, medium: { url: '' } }, title: 'title', viewCount: 1,
};

jest.mock('../../context/KeyboardContext', () => ({
  useKeyboardContext: () => ({
    keyboard: () => mockedKeyboard(),
    dispatch: () => mockedDispatch(),
  }),
}));

describe('<Search /> component', () => {
  const component = renderer.create(<VideoCard data={mockVideoItem} position={1} activeItem={1} />);

  const jsonComponent: ReactTestRendererJSON | any = component.toJSON();

  it('should render properly', () => {
    expect(jsonComponent).toMatchSnapshot();
  });

  it('should render container div component', () => {
    const tree = shallow(<VideoCard data={mockVideoItem} position={1} activeItem={1} />).dive();

    expect(tree.exists('div')).toBeTruthy();
  });

  it('should render elements of VideoCard', () => {
    expect(jsonComponent.children).toHaveLength(4);
  });
});
