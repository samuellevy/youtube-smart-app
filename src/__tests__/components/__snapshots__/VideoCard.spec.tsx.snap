// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> component should render properly 1`] = `
<div
  className="sc-bdfBwQ crPqTE"
>
  <div
    className="sc-gsTCUz ldhubU"
  >
    <img
      alt=""
      className="sc-dlfnbm bevxTP"
      src=""
    />
  </div>
  <h2
    className="sc-hKgILt"
  >
    title
  </h2>
  <div
    className="sc-eCssSg kZBOjq"
  >
    teste
  </div>
  <div
    className="sc-jSgupP kvgBhb"
  >
    <div
      className="sc-gKsewC"
    >
      1 views â€¢ published
    </div>
  </div>
</div>
`;
